File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    answer = ''
    for i in range(1,len(food)) :
        if food[i]//2 > 0 :
            for j in range(food[i]//2) :
                answer += str(i)
    answer +='0'
    for i in range(len(food)-1,-1,-1) :
        if food[i]//2 > 0 :
            for j in range(food[i]//2) :
                answer += str(i)
            
    return answer

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public StringBuilder solution(int[] food) {
        StringBuilder answer = new StringBuilder();
        StringBuilder real_answer = new StringBuilder();
        
        for(int i =1; i<food.length; i++){
            if (food[i]%2 != 0 ){
                food[i] -=1;
            }else {
                
            }
            for (int j=0; j<food[i]/2; j++){
                answer.append(i);
            }
        }
        real_answer.append(answer);
        real_answer.append('0');
        StringBuilder ans_rev = answer.reverse();
        real_answer.append(ans_rev);
 
        return real_answer;
    }
}